core/modules/qproc/QuerySession.h

```diff
@@ -79,7 +78,17 @@ class QuerySession {
 
     std::string const& getOriginal() const { return _original; }
     void setDefaultDb(std::string const& db);
-    void setQuery(std::string const& q);
+
+    /**
+     * @brief Analyze SQL query issued by user
+     *
+     * This query comes from user through mysql-client and mysql-proxy
+     * This function will parse it, apply query plugins (i.e. build parallel and
+     * merge queries) and check for errors
+     *
+     * @param sql: the sql query
+     */
+    void analyzeQuery(std::string const& sql);
```

Comment:
`analyzeQuery` feels misleading to me. The method does not return anything, so I presume it just parses the query and updates internals of this object with that parsed query. `setQuery` looks better if that is true.